packages/datadog-plugin-aws-sdk/test/serverless-peer-service.spec.js

```diff
@@ -0,0 +1,319 @@
+'use strict'
+
+const agent = require('../../dd-trace/test/plugins/agent')
+const Config = require('../../dd-trace/src/config')
+const helpers = require('./kinesis_helpers')
+const { promisify } = require('util')
+const { setup } = require('./spec_helpers')
+const sinon = require('sinon')
+
+describe('Plugin', () => {
+  describe('Serverless', function () {
+    let tracer
+    setup()
+
+    withVersions('aws-sdk', ['aws-sdk', '@aws-sdk/smithy-client'], (version, moduleName) => {
+      let AWS
+
+      before(async () => {
+        tracer = require('../../dd-trace')
+        sinon.stub(Config.prototype, '_isInServerlessEnvironment').returns(true)
+        // force a reload of the plugins by deleting the plugin instance
+        delete tracer._pluginManager._pluginsByName['aws-sdk']
```

Comment:
Why is this needed?

Reply:
The config object is only read at the time of loading, so we need to induce a reload by deleting the plugin. That way the newly changed Config var is read at the proper time.